canonical-web-and-design/maas-ui
# feat(ui): add ability to create datastores

## Done

- Added the ability to create datastores
- Made the datastores table dependent on the storage layout, since it's possible to be in VMFS6 storage layout without any datastores

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a Ready or Allocated machine
- Change the storage layout to VMFS6
- If the machine only has one disk, remove the datastore that has been created
- Tick the checkbox for the released partition and check that the "Create datastore" button is enabled
- Click "Create datastore" and change the name
- Submit the form and check that the datastore is successfully created and shows in the datastores table 

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2169

## Screenshot

![Screenshot_2021-01-21 sure-cod maas storage bolla MAAS](https://user-images.githubusercontent.com/25733845/105287658-6723f880-5c03-11eb-94bf-7525969b1229.png)
